EventListener/ParamConverterListener.php

```diff
@@ -96,7 +96,7 @@ private function autoConfigure(\ReflectionFunctionAbstract $r, Request $request,
                 $configurations[$name]->setClass($param->getClass()->getName());
             }
 
-            $configurations[$name]->setIsOptional($param->isOptional());
+            $configurations[$name]->setIsOptional($param->isOptional() || method_exists($param, 'hasType') && $param->hasType() && $param->getType()->allowsNull());
```

Comment:
Can you add the `method_exists($param, 'hasType')` to the constructor and store it in a property? Something like added here: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php#L56

```php
$this->supportsScalarTypes = method_exists('ReflectionParameter', 'getType');
```